src/worker/jobs/downloads/update_metadata.rs

```diff
@@ -145,12 +145,19 @@ fn batch_update(batch_size: i64, conn: &mut PgConnection) -> QueryResult<i64> {
                     FROM downloads_batch
                 ) sum
                 WHERE sum.downloads > 0
+            ), sorted_downloads_batch AS (
+                -- Sort the `downloads_batch` CTE by `version_id` and `date` to
+                -- ensure that the `version_downloads` table is updated in a
+                -- consistent order to avoid deadlocks.
+                SELECT version_id, date, downloads FROM downloads_batch
+                ORDER BY version_id, date
```

Comment:
We might need a `SELECT ORDER BY FOR UPDATE` to guarantee lock order.
Something like:
```sql
SELECT downloads_batch.*
FROM version_downloads
JOIN downloads_batch using (version_id, date)
ORDER BY version_id, date
FOR UPDATE
```